TSPL.docc/LanguageGuide/Protocols.md

```diff
@@ -749,124 +755,60 @@ a nonfailable initializer or an implicitly unwrapped failable initializer.
 ## Protocols as Types
 
 Protocols don't actually implement any functionality themselves.
-Nonetheless,
-you can use protocols as a fully fledged types in your code.
-Using a protocol as a type is sometimes called an *existential type*,
-which comes from the phrase
-"there exists a type *T* such that *T* conforms to the protocol".
-
-You can use a protocol in many places where other types are allowed, including:
-
-- As a parameter type or return type in a function, method, or initializer
-- As the type of a constant, variable, or property
-- As the type of items in an array, dictionary, or other container
-
-> Note: Because protocols are types,
-> begin their names with a capital letter
-> (such as `FullyNamed` and `RandomNumberGenerator`)
-> to match the names of other types in Swift
-> (such as `Int`, `String`, and `Double`).
-
-Here's an example of a protocol used as a type:
-
-```swift
-class Dice {
-    let sides: Int
-    let generator: RandomNumberGenerator
-    init(sides: Int, generator: RandomNumberGenerator) {
-        self.sides = sides
-        self.generator = generator
-    }
-    func roll() -> Int {
-        return Int(generator.random() * Double(sides)) + 1
-    }
-}
-```
-
-<!--
-  - test: `protocols`
+Nonetheless, you can use a protocol as a type in your code.
 
-  ```swifttest
-  -> class Dice {
-        let sides: Int
-        let generator: RandomNumberGenerator
-        init(sides: Int, generator: RandomNumberGenerator) {
-           self.sides = sides
-           self.generator = generator
-        }
-        func roll() -> Int {
-           return Int(generator.random() * Double(sides)) + 1
-        }
-     }
-  ```
+The most common way to use a protocol as a type
+is to use a protocol as a generic constraint.
+Code with generic constraints can work with
+any type that conforms to the protocol,
+and the specific type is chosen by the code that uses the API.
+For example,
+when you call a function that takes an argument
+and that argument's type is generic,
+the caller chooses the type.
+
+Code with an opaque type
+works with some type that conforms to the protocol.
+The underlying type is known at compile time,
+and the API implementation chooses that type,
+but that type's identity is hidden from clients of the API.
+Using an opaque type lets you prevent implementation details of an API
+from leaking through the layer of abstraction ---
+for example, by hiding the specific return type from a function,
+and only guaranteeing that the value conforms to a given protocol.
+
+Code with a boxed protocol type
+works with any type, chosen at runtime, that conforms to the protocol.
+To support this runtime flexibility,
+Swift adds a level of indirection when necessary ---
+this indirection is known as a *box*,
+and it has a performance cost.
+Because of this flexibility,
+Swift doesn't know the underlying type at compile time,
+which means you can access only the members
```

Comment:
```suggestion
which means you can only access the members
```

Reply:
Just to confirm — normally, "only" goes immediately before the specific word that it modifies, to avoid ambiguity. I'm assuming in this case, you'd move it in front of the verb to follow a more spoken-English style, because there isn't another action that you can perform on a member — so it's unlikely to be misread with "only" modifying the verb.

Reply:
Feel free to stet if you think this changes the meaning.